### ISSUE TITLE
False positive for `foo="<span class="bar">baz</span>"`

### ISSUE THREAD

**Issue by gharlan** [voku/anti-xss#85]
Input:

```html
<span>
    foo="<span class="bar">baz</span>"
</span>
```

Expected result (no changes):

```html
<span>
    foo="<span class="bar">baz</span>"
</span>
```

Actual result:

```html
<span>
    foo="&lt;span class="bar">baz</span>"
</span>
```

The change is coming from this part: https://github.com/voku/anti-xss/blob/19da849cb2dd44d7c25e3eb0ee6cc9433ff9a106/src/voku/helper/AntiXSS.php#L1543-L1559

**Comment by gharlan** [comment]
The part above is changing the code, but the cause seems to be this regex:
https://github.com/voku/anti-xss/blob/19da849cb2dd44d7c25e3eb0ee6cc9433ff9a106/src/voku/helper/AntiXSS.php#L539

It detects `foo="..."` as an attribute, but actually it is content of the outer span tag.

**Comment by voku** [comment]
Hi, thanks for the bug report. 

I already looked into it and the problem is that we use `$regExForHtmlTags = '/<\p{L}+.*+/us';` to decode the HTML attributes, but because we want to decode also HTML attributes in broken tags, I thought it's a good idea to look only for the open tags. :-/

If we replace it with `$regExForHtmlTags = '/<\p{L}+.*>/usU';`, then it's working, but I need to check what this is changing for all the test cases.

**Comment by gharlan** [comment]
Maybe something like https://regex101.com/r/loBzsU/1?

**Comment by voku** [comment]
Thanks for the fix: :) 

⇒ fixed in version 4.1.34
